### ISSUE TITLE
[Editor] Dragging objects into the scene should snap them to existing geometry

### ISSUE THREAD

**Issue by darrylryan** [AtomicGameEngine/AtomicGameEngine#1406]
If we drag a character onto the terrain, or onto a platform in the scene, the editor should use a ray pick to spawn him on the actual surface we're pointing to and not simply floating somewhere in front of the mouse's screen coordinates.

**Comment by Alan-FGR** [comment]
I agree. This is kinda important. We could just use the object origin like the other engines but placement is very important for level editing, in the future we may want to have some smart placement tools like offsets and using the geometry (so even if the origin isn't at the base of the mesh we offset it by the lowest vertex), and for prefabs we could optionally use colliders and whatnot. We'd need a decent UI to quickly an easily choose the mode you want and maybe use modifier keys (e.g: shift+drag uses geometry).
